h5io/_h5io.py

```diff
@@ -893,7 +909,11 @@ def _setstate(obj_class, state_dict):
     else:
         args = ()
         kwargs = {}
-    obj = obj_class.__new__(obj_class, *args, **kwargs)
+    if inspect.isclass(obj_class):
+        obj = obj_class.__new__(obj_class, *args, **kwargs)
+    else:
+        # We got a singleton-like object
```

Comment:
```suggestion
        # We got an object which is not a class - it could be a singleton-like object - we just return the object without initialisation. 
```